src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -149,6 +164,47 @@ private UUID publishRequest(UUID consortiumId, PublicationRequest publicationReq
     return null;
   }
 
+  private void getFailedSettingsForFailedTenants(UUID consortiumId, UUID publicationId, SharingSettingRequest sharingSettingRequest) {
+    // NOSONAR
+    boolean desiredResultReceived = false;
+    long startTime = System.currentTimeMillis();
+    long timeout = 3000;
+
+    while (Boolean.FALSE.equals(desiredResultReceived) && System.currentTimeMillis() - startTime < timeout) {
+      var publicationStatusEntity = publicationStatusRepository.findById(publicationId);
+      if (publicationStatusEntity.isPresent()) {
+        PublicationDetailsResponse publicationDetails = publicationService.getPublicationDetails(consortiumId, publicationId);
+        if (publicationDetails.getStatus() == PublicationStatus.ERROR) {
+          desiredResultReceived = true;
+
+          List<PublicationResult> publicationResults = publicationService
+            .getPublicationResults(consortiumId, publicationId)
+            .getPublicationResults()
+            .stream()
+            .filter(publicationResult -> ObjectUtils.notEqual(publicationResult.getResponse(), NULL)).toList();
+
+          updateFailedSettingsToLocal(consortiumId, sharingSettingRequest, publicationResults);
+        }
+      }
+    }
+  }
+
+  private void updateFailedSettingsToLocal(UUID consortiumId, SharingSettingRequest sharingSettingRequest, List<PublicationResult> publicationResults) {
+    // NOSONAR
+    JsonNode payload = objectMapper.convertValue(sharingSettingRequest.getPayload(), JsonNode.class);
+    var updatedPayload = ((ObjectNode) payload).set(SOURCE, new TextNode(LOCAL_SETTING_SOURCE));
+
+    publicationResults.forEach(publicationResult -> {
+      PublicationRequest publicationPutRequest = createPublicationRequestForSetting(sharingSettingRequest, HttpMethod.PUT.toString());
+      publicationPutRequest.setPayload(updatedPayload);
+      publicationPutRequest.setTenants(Set.of(publicationResult.getTenantId()));
+
+      try (var ignored = new FolioExecutionContextSetter(contextHelper.getSystemUserFolioExecutionContext(folioExecutionContext.getTenantId()))) {
```

Comment:
If for example we have 10 failed tenants - we don't need to initialize 10 publish coordinator requests, we need only 1 requests that has 10 tenants

Reply:
fixed